lms/djangoapps/course_goals/handlers.py

```diff
@@ -3,25 +3,37 @@
 """
 
 
-from django.db import models
+from django.db.models.signals import post_save
 from django.dispatch import receiver
+from eventtracking import tracker
 
 from common.djangoapps.course_modes.models import CourseMode
 from common.djangoapps.student.models import CourseEnrollment
+from common.djangoapps.track import segment
 
-from .api import add_course_goal, remove_course_goal
-from .models import GOAL_KEY_CHOICES
+from .api import add_course_goal
+from .models import GOAL_KEY_CHOICES, CourseGoal
 
 
-@receiver(models.signals.post_save, sender=CourseEnrollment, dispatch_uid="update_course_goal_on_enroll_change")
+@receiver(post_save, sender=CourseEnrollment, dispatch_uid="update_course_goal_on_enroll_change")
 def update_course_goal_on_enroll_change(sender, instance, **kwargs):  # pylint: disable=unused-argument
     """
     Updates goals as follows on enrollment changes:
     1) Set the course goal to 'certify' when the user enrolls as a verified user.
-    2) Remove the course goal when the user's enrollment is no longer active.
     """
     course_id = str(instance.course_id)
-    if not instance.is_active:
-        remove_course_goal(instance.user, course_id)
-    elif instance.mode == CourseMode.VERIFIED:
-        add_course_goal(instance.user, course_id, GOAL_KEY_CHOICES.certify)
+    if instance.mode == CourseMode.VERIFIED:
+        add_course_goal(instance.user, course_id, GOAL_KEY_CHOICES.certify, manually_updated=False)
```

Comment:
i see manually_updated is passed in as False here, I don't see anywhere where it is passed in as True?

Reply:
I may have done a late update to this PR. It's passed in as True everywhere else this function is used as well as all of the spots a CourseGoal is created using the object manager directly.